### PR TITLE
feat(chat): display git repo connected status in chat side panel

### PR DESCRIPTION
**Pull request by liangfung**
### Screen record
https://jam.dev/c/78d7ffb9-434f-40eb-89cf-d2ac4f60e2d5

### CODE REVIEW

File: ee/tabby-ui/components/chat/repo-select.tsx [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+                <span
+                  className={cn('truncate', {
+                    'text-muted-foreground': !selectedRepoName
+                  })}
+                >
+                  {selectedRepoName || 'Select repo...'}
+                </span>
+                {isWorkspaceRepo && (
+                  <span className="shrink-0 text-muted-foreground">
+                    {isExtraSmallScreen ? 'Workspace' : 'Repo in workspace'}
```
Comment (wsxiaoys) [inline ee/tabby-ui/components/chat/repo-select.tsx:79]:
```suggestion
                    Context
```

File: ee/tabby-ui/components/chat/repo-select.tsx [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+                    'text-muted-foreground': !selectedRepoName
+                  })}
+                >
+                  {selectedRepoName || 'Select repo...'}
+                </span>
+                {isWorkspaceRepo && (
+                  <span className="shrink-0 text-muted-foreground">
+                    Context
+                  </span>
+                )}
```
Comment (wsxiaoys) [inline ee/tabby-ui/components/chat/repo-select.tsx:81]:
```suggestion
```

File: ee/tabby-ui/components/chat/repo-select.tsx [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+          <DropdownMenuTrigger className="outline-none" asChild>
+            <div className="flex flex-1 cursor-pointer items-center gap-1.5 overflow-hidden">
+              <IconFolderGit className="shrink-0" />
+              <div className="flex flex-1 items-center gap-1.5 truncate break-all">
+                <span
+                  className={cn('truncate', {
+                    'text-muted-foreground': !selectedRepoName
+                  })}
+                >
+                  {selectedRepoName || 'Select repo...'}
```
Comment (wsxiaoys) [inline ee/tabby-ui/components/chat/repo-select.tsx:75]:
```suggestion
                  {selectedRepoName || 'Workspace'}
```

File: ee/tabby-ui/components/chat/repo-select.tsx [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+                          {repo.sourceName}
+                        </div>
+                      </div>
+                      {repo.sourceId === workspaceRepoId && (
+                        <span className="ml-1.5 shrink-0 text-muted-foreground">
+                          {isExtraSmallScreen
+                            ? 'Workspace'
+                            : 'Repo in workspace'}
+                        </span>
+                      )}
```
Comment (wsxiaoys) [inline ee/tabby-ui/components/chat/repo-select.tsx:190]:
```suggestion
```

File: clients/tabby-chat-panel/src/index.ts [hunk 5]
```diff
@@ -224,6 +231,9 @@ export interface ClientApiMethods {
    * @returns Whether the file location is opened successfully.
    */
   openInEditor: (target: FileLocation) => Promise<boolean>
+
+  // Provide all repos found in workspace folders.
+  provideWorkspaceGitRepoInfo?: () => Promise<GitRepoInfo[]>
```
Comment (wsxiaoys) [inline clients/tabby-chat-panel/src/index.ts:236]:
```suggestion
  readWorkspaceGitRepositories?: () => Promise<GitRepoInfo[]>
```

File: clients/tabby-chat-panel/src/index.ts [hunk 6]
```diff
@@ -177,6 +177,13 @@ export interface SymbolInfo {
   target: FileLocation
 }
 
+/**
+ * Includes information about a git repository in workspace folder
+ */
+export interface GitRepoInfo {
```
Comment (wsxiaoys) [inline clients/tabby-chat-panel/src/index.ts:183]:
```suggestion
export interface GitRepository {
```

File: clients/tabby-chat-panel/src/index.ts [hunk 7]
```diff
@@ -177,6 +177,13 @@ export interface SymbolInfo {
...
+  gitUrl: string
```
Comment (wsxiaoys) [inline clients/tabby-chat-panel/src/index.ts:184]:
```suggestion
  url: string
```

File: ee/tabby-schema/src/schema/mod.rs [hunk 8]
```diff
@@ -549,8 +549,32 @@ impl Query {
             .await
     }
 
+    async fn resolve_git_url(ctx: &Context, git_url: String) -> Result<Option<Repository>> {
```
Comment (wsxiaoys) [inline ee/tabby-schema/src/schema/mod.rs:552]: can this logic be done in client side? e.g with fuzzy match logic

File: clients/tabby-chat-panel/src/index.ts [hunk 9]
```diff
@@ -224,6 +231,9 @@ export interface ClientApiMethods {
    * @returns Whether the file location is opened successfully.
    */
   openInEditor: (target: FileLocation) => Promise<boolean>
+
+  // Provide all repos found in workspace folders.
+  readWorkspaceGitRepositories?: () => Promise<GitRepository[]>
```
Comment (wsxiaoys) [inline clients/tabby-chat-panel/src/index.ts:236]:
```typesceript
findClosestGitRepository(workspace: GitRepositories[], gitUrl: String);
```

**Review by wsxiaoys — APPROVED** [review]

**Review by icycodes — APPROVED** [review]

**Pull request merged**